source/common/network/proxy_protocol.cc

```diff
@@ -143,11 +139,11 @@ bool ProxyProtocol::ActiveConnection::readLine(int fd, std::string& s) {
       }
     }
 
+    // Read the data upto and including the line feed, if available, but not past it.
+    // This should never fail, as search_index_ - buf_off_ <= nread, so we're asking
+    // only for bytes we have already seen.
     nread = recv(fd, buf_ + buf_off_, search_index_ - buf_off_, 0);
-
-    if (nread < 1) {
-      throw EnvoyException("failed to read proxy protocol");
-    }
+    ASSERT(size_t(nread) == search_index_ - buf_off_);
```

Comment:
It would take quite a severe OS bug for recv() to fail getting data already received with the PEEK flag. If you think this risk is real, then I'll revert this change with the cost of not having 100% coverage. Please advice.